app/src/main/java/org/oppia/android/app/options/AppLanguageFragment.kt

```diff
@@ -7,29 +7,44 @@ import android.view.View
 import android.view.ViewGroup
 import org.oppia.android.app.fragment.FragmentComponentImpl
 import org.oppia.android.app.fragment.InjectableFragment
-import org.oppia.android.util.extensions.getStringFromBundle
+import org.oppia.android.app.model.AppLanguageFragmentArguments
+import org.oppia.android.app.model.AppLanguageFragmentStateBundle
+import org.oppia.android.app.model.OppiaLanguage
+import org.oppia.android.util.extensions.getProto
+import org.oppia.android.util.extensions.putProto
 import javax.inject.Inject
 
-private const val APP_LANGUAGE_PREFERENCE_TITLE_ARGUMENT_KEY =
-  "AppLanguageFragment.app_language_preference_title"
-private const val APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_ARGUMENT_KEY =
-  "AppLanguageFragment.app_language_preference_summary_value"
-private const val SELECTED_LANGUAGE_SAVED_KEY = "AppLanguageFragment.selected_language"
-
 /** The fragment to change the language of the app. */
 class AppLanguageFragment : InjectableFragment(), AppLanguageRadioButtonListener {
 
   @Inject
   lateinit var appLanguageFragmentPresenter: AppLanguageFragmentPresenter
 
   companion object {
-    fun newInstance(prefsKey: String, prefsSummaryValue: String): AppLanguageFragment {
-      val fragment = AppLanguageFragment()
-      val args = Bundle()
-      args.putString(APP_LANGUAGE_PREFERENCE_TITLE_ARGUMENT_KEY, prefsKey)
-      args.putString(APP_LANGUAGE_PREFERENCE_SUMMARY_VALUE_ARGUMENT_KEY, prefsSummaryValue)
-      fragment.arguments = args
-      return fragment
+    private const val FRAGMENT_ARGUMENTS_KEY = "AppLanguageFragment.arguments"
+    private const val FRAGMENT_SAVED_STATE_KEY = "AppLanguageFragment.saved_state"
+
+    fun newInstance(prefsSummaryValue: OppiaLanguage): AppLanguageFragment {
```

Comment:
Ditto here--please update the name accordingly.

Reply:
Updated.